project-base/storefront/components/Pages/Order/PaymentConfirmation/Gateways/GoPayGateway.tsx

```diff
@@ -20,22 +20,20 @@ export const GoPayGateway: FC<GoPayGatewayProps> = ({ orderUuid, requiresAction,
     const wasPaidRef = useRef(false);
 
     useEffect(() => {
-        if (wasPaidRef.current || !initiatedPaymentGate) {
-            return;
-        }
+        if (!wasPaidRef.current && initiatedPaymentGate) {
+            payOrder({ orderUuid }).then((payOrderResult) => {
+                if (payOrderResult.error?.graphQLErrors !== undefined) {
+                    for (const error of payOrderResult.error.graphQLErrors) {
+                        showErrorMessage(error.message);
+                    }
 
-        payOrder({ orderUuid }).then((payOrderResult) => {
-            if (payOrderResult.error?.graphQLErrors !== undefined) {
-                for (const error of payOrderResult.error.graphQLErrors) {
-                    showErrorMessage(error.message);
+                    return;
                 }
 
-                return;
-            }
-
-            setGoPayPaymentSetup(payOrderResult.data?.PayOrder.goPayCreatePaymentSetup ?? undefined);
-        });
-        wasPaidRef.current = true;
+                setGoPayPaymentSetup(payOrderResult.data?.PayOrder.goPayCreatePaymentSetup ?? undefined);
+            });
+            wasPaidRef.current = true;
```

Comment:
Why do we need ref here?

Reply:
It creates a payment transaction on BE every time it is called (the `payOrder` mutation). This way it is only called once.